bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeBridgeHandler.java

```diff
@@ -0,0 +1,819 @@
+
+            while (iter.hasNext()) {
+                String eventDatapointID = iter.next();
+
+                JsonElement element = jsonObject.get(eventDatapointID);
+                String value = element.getAsString();
+
+                channelUpdateHandler.updateChannelByDatapointEvent(eventDatapointID, value);
+
+                logger.debug("Socket event processed: event-datapoint-ID {} value {}", eventDatapointID, value);
+            }
+        }
+    }
+
+    /**
+     * Method to open Http connection
+     *
+     * @author Andras Uhrin
+     *
+     */
+    public boolean openHttpConnection() {
+        boolean ret = false;
+
+        // Configure client
+        httpClient.setFollowRedirects(false);
+        httpClient.setMaxConnectionsPerDestination(1);
+        httpClient.setMaxRequestsQueuedPerDestination(50);
+
+        // Set timeouts
+        httpClient.setIdleTimeout(-1);
+        httpClient.setConnectTimeout(5000);
+
+        try {
+            // Start HttpClient.
+            switch (httpClient.getState()) {
+                case AbstractLifeCycle.FAILED:
+                case AbstractLifeCycle.STOPPED: {
+                    httpClient.start();
+                    break;
+                }
+                case AbstractLifeCycle.STARTING:
+                case AbstractLifeCycle.STARTED:
+                case AbstractLifeCycle.STOPPING: {
+                    // nothing to do
+                    break;
+                }
+            }
+
+            logger.info("Start http client");
+
+            ret = true;
+        } catch (Exception ex) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Cannot start http client");
+
+            logger.error("Cannot start http client - {}", ex.getMessage());
+
+            ret = false;
+        }
+
+        if (ret) {
+            // Add authentication credentials and make a check.
+            try {
+                // Add authentication credentials.
+                AuthenticationStore auth = httpClient.getAuthenticationStore();
+
+                URI uri1 = new URI("http://" + ipAddress + "/fhapi/v1");
+                auth.addAuthenticationResult(new BasicAuthentication.BasicResult(uri1, username, password));
+
+                String url = "http://" + ipAddress + "/fhapi/v1/api/rest/devicelist";
+
+                Request req = httpClient.newRequest(url);
+                ContentResponse res = req.send();
+
+                // check status
+                if (res.getStatus() == 200) {
+                    // response OK
+                    httpConnectionOK.set(true);
+
+                    ret = true;
+
+                    logger.info("HTTP connection to SysAP is OK");
+                } else {
+                    // response NOK, set error
+                    httpConnectionOK.set(false);
+
+                    ret = false;
+
+                    logger.info("Wrong credentials for SysAP");
```

Comment:
```suggestion
                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Wrong credentials for SysAP");
```

or better yet:
```suggestion
                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/offline.conf-error.wrong-credentials");
```

with corresponding text in `freeathome.properties` for I18N support.

Reply:
suggested change with localisable text is taken over.